lib/assert.js

```diff
@@ -448,13 +448,24 @@ function getActual(block) {
   return NO_EXCEPTION_SENTINEL;
 }
 
+function checkIsPromise(obj) {
+  return isPromise(obj) ||
+    obj !== null && typeof obj === 'object' &&
+    typeof obj.then === 'function' &&
+    typeof obj.catch === 'function';
+}
+
 async function waitForActual(block) {
-  if (typeof block !== 'function') {
-    throw new ERR_INVALID_ARG_TYPE('block', 'Function', block);
+  let resultPromise;
+  if (typeof block === 'function') {
+    // Return a rejected promise if `block` throws synchronously.
+    resultPromise = block();
```

Comment:
Should we check that `block()` effectively returns a Promise?

Reply:
I have a follow up PR in #19886 that does exactly that :-)